src/execution_plan/execution_plan_modify.c

```diff
@@ -264,6 +264,10 @@ void ExecutionPlan_BoundVariables(const OpBase *op, rax *modifiers) {
 		}
 	}
 
+	/* Project and Aggregate operations demarcate variable scopes,
+	 * collect their projections but do not recurse into their children. */
+	if(op->type == OPType_PROJECT || op->type == OPType_AGGREGATE) return;
+
```

Comment:
Simple enough!
If we were to introduce any new operation(s) which dictates scope boundaries, it would be easy to overlook this condition.

Reply:
Added a comment to this effect.